### PR TITLE
HCAP-1513 allow MoH to involve in the hire process

### PR DESCRIPTION
**Pull request by sunghwan-park-fw**
- Allow MoH user to change `interested` for hired applicants
- Show 'Candidates,' 'Archived Candidates,' 'Hired Candidates,' and 'Return of Service' tabs to MoH users
- Show applicant's status in the hiring process
- Allow MoH to update applicant's status
- Add all PEOI answer fields to test data

### CODE REVIEW

File: server/scripts/randomize-participants.ts [hunk 1]
```diff
@@ -1,18 +1,46 @@
 import { faker } from '@faker-js/faker';
 import _ from 'lodash';
 import { convertToCsv } from './services/participant-seed';
-import { Program } from '../constants';
+import { healthRegions, Program, yesOrNo } from '../constants';
 
 // this is for canadian postal codes
 faker.locale = 'en_CA';
 
-const DEFAULT_PREF_LOCATION = 'Fraser';
```
Comment (coryatwater-FW) [inline server/scripts/randomize-participants.ts:9]: I remember we used `Fraser` for a reason, don't remember if it was just for simplicity's sake though

File: server/scripts/randomize-participants.ts [hunk 2]
```diff
@@ -30,10 +58,16 @@ const generateParticipants = async (amount: number) => {
         postalCodeFsa: pc.slice(0, 3),
         phoneNumber: faker.phone.number('##########'),
         emailAddress: faker.internet.email(fn, ln),
-        preferredLocation: DEFAULT_PREF_LOCATION,
+        preferredLocation: _.sample(healthRegions),
         interested: 'yes',
         crcClear: 'yes',
         program: _.sample([Program.HCA, Program.MHAW]),
```
Comment (coryatwater-FW) [inline server/scripts/randomize-participants.ts:64]: are we wanting to filter out HA's that aren't doing the MHAW path at this time?

File: client/src/components/modal-forms/ProspectingForm.js [hunk 3]
```diff
@@ -18,30 +20,41 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
+const getConfirmMessage = (hasMultipleParticipants, isMoH) => {
+  let message = hasMultipleParticipants ? ` participants have` : ' has';
+  message += ` been engaged.`;
+  if (!isMoH) {
+    message += 'These candidates can now be found in the My Candidates Tab.';
```
Comment (coryatwater-FW) [inline client/src/components/modal-forms/ProspectingForm.js:27]: add a few spaces for after the `.`?

File: client/src/constants/participantTableConstants.js [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import ToastStatus from './toast';
 import { Role } from './user-roles';
 import { Program } from './programs';
+import _ from 'lodash';
```
Comment (coryatwater-FW) [inline client/src/constants/participantTableConstants.js:4]: import to top?

File: server/scripts/randomize-participants-status.ts [hunk 5]
```diff
@@ -53,19 +54,16 @@ const positionType = ['casual', 'permanent'];
 const employmentType = ['part-time', 'full-time'];
 
 const randomize = (n: number, reject?: boolean) => {
-  let value = 0;
+  let value = _.random(1, n);
   // prevent duplicate participant ids from being used
   // Sonarcloud complains about Math.random, ignoring as it's just being used for simple random numbers
```
Comment (coryatwater-FW) [inline server/scripts/randomize-participants-status.ts:59]: can probably remove this comment now

File: server/scripts/randomize-participants.ts (hunk 1 above)
Comment (sunghwan-park-fw) [reply server/scripts/randomize-participants.ts:9]: You're right. It caused a lot of problems, but I just fixed the participants.

File: server/services/participants/participant-entries.ts [hunk 6]
```diff
@@ -282,22 +275,7 @@ export const getParticipants = async (
...
-        driverLicense: item.driverLicense,
-        experienceWithMentalHealthOrSubstanceUse: item.experienceWithMentalHealthOrSubstanceUse,
-        preferredLocation: item.preferredLocation,
-        currentOrMostRecentIndustry: item.currentOrMostRecentIndustry,
-        roleInvolvesMentalHealthOrSubstanceUse: item.roleInvolvesMentalHealthOrSubstanceUse,
-        nonHCAP: item.nonHCAP,
-        userUpdatedAt: item.userUpdatedAt,
-        callbackStatus: item.callbackStatus,
-        distance: item.distance,
+        ..._.pick(item, user.isMoH ? participantFieldsForSuper : participantFieldsForEmployer),
```
Comment (coryatwater-FW) [inline server/services/participants/participant-entries.ts:278]: should this be `participantFieldsForMoH`?

Comment (sunghwan-park-fw) [reply server/services/participants/participant-entries.ts:278]: I missed your comment. Just fixed it.

File: .docker/keycloak/realm-testing.json [hunk 7]
```diff
@@ -1088,9 +1088,9 @@
       "clientAuthenticatorType": "client-secret",
       "secret": "**********",
       "redirectUris": [
-        "http://hcapemployers.local.freshworks.club:4000/*",
+        "http://hcapemployers.localhost:4000/*",
```
Comment (fw-noel) [inline .docker/keycloak/realm-testing.json:1091]: I imagine these will have to be changed in the readme as well

**Review by fw-noel — APPROVED** [review]

**Review by fw-noel — APPROVED** [review]

**Pull request merged**